solr/core/src/java/org/apache/solr/cloud/CreateAliasCmd.java

```diff
@@ -155,60 +104,61 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results)
     final String aliasName = message.getStr(CommonParams.NAME);
     ZkStateReader zkStateReader = ocmh.zkStateReader;
     ZkStateReader.AliasesManager holder = zkStateReader.aliasesHolder;
+
+    //TODO refactor callCreatePlainAlias
     if (!anyRoutingParams(message)) {
+
       final List<String> canonicalCollectionList = parseCollectionsParameter(message.get("collections"));
       final String canonicalCollectionsString = StrUtils.join(canonicalCollectionList, ',');
       validateAllCollectionsExistAndNoDups(canonicalCollectionList, zkStateReader);
       holder.applyModificationAndExportToZk(aliases -> aliases.cloneWithCollectionAlias(aliasName, canonicalCollectionsString));
-    } else {
-      final String routedField = message.getStr(ROUTING_FIELD);
+
+    } else { //TODO refactor callCreateRoutedAlias
+
+      // Validate we got everything we need
+      if (!message.getProperties().keySet().containsAll(REQUIRED_ROUTING_PARAMS)) {
+        throw new SolrException(BAD_REQUEST, "A routed alias requires these params: " + REQUIRED_ROUTING_PARAMS
+        + " plus some create-collection prefixed ones.");
+      }
+
+      Map<String, String> aliasMetadata = new LinkedHashMap<>();
+      message.getProperties().entrySet().stream()
+          .filter(entry -> PARAM_IS_METADATA.test(entry.getKey()))
+          .forEach(entry -> aliasMetadata.put(entry.getKey(), (String) entry.getValue()));
+
+      //TODO read these from metadata where appropriate. This leads to consistent logic between initial routed alias
+      //  collection creation, and subsequent collections to be created.
+
       final String routingType = message.getStr(ROUTING_TYPE);
       final String tz = message.getStr(TZ);
       final String start = message.getStr(START);
       final String increment = message.getStr(ROUTING_INCREMENT);
       final String maxFutureMs = message.getStr(ROUTING_MAX_FUTURE);
 
       try {
-        if (0 > Long.parseLong(maxFutureMs)) {
+        if (maxFutureMs != null && 0 > Long.parseLong(maxFutureMs)) {
```

Comment:
I think this means you want maxFutureMs to be required? Should it be added to REQUIRED_ROUTING_PARAMS?

Reply:
oh no it's optional; I was making it optional here.  The code as I saw it (and tested) would fail if it wasn't present, and I want it optional.

Reply:
oh right I see. I read it too quick. Yes, need to defend the parseLong from null, I agree.